Mathlib/RingTheory/Smooth/StandardSmooth.lean

```diff
@@ -0,0 +1,208 @@
+/-
+Copyright (c) 2024 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jung Tao Cheng, Christian Merten, Andrew Yang
+-/
+import Mathlib.Algebra.MvPolynomial.PDeriv
+import Mathlib.LinearAlgebra.Determinant
+import Mathlib.RingTheory.Presentation
+
+/-!
+# Standard smooth algebras
+
+In this file we define standard smooth presentations for algebras. This notion
+is introduced for `SubmersivePresentations`, which is a `Presentation`
+that has less relations than generators. More precisely there exists
+an injective map from `P.relations` to `P.vars`.
+
+Finally, we introduce the notion of standard smooth algebras, which are algebras that
+admit a standard smooth presentation.
+
+## Main definitions
+
+All of these are in the `Algebra` namespace. Let `S` be an `R`-algebra.
+
+- `SubmersivePresentation`: A `Presentation` of `S` as `R`-algebra, equipped with an injective map
+  from `P.relations` to `P.vars`. This map is used to define the differential of a submersive
+  presentation.
+
+Let now `P` be a submersive presentation of `S` over `R`.
+
+- `SubmersivePresentation.differential`: A linear endomorphism of `P.rels → P.ring` sending
+  the `j`-th standard basis vector, corresponding to the `j`-th relation, to the vector
+  of partial derivatives of `P.relation j` with respect to the coordinates `P.map i` for
+  `i : P.rels`.
+- `SubmersivePresentation.jacobian`: The determinant of `P.differential`.
+- `SubmersivePresentation.jacobiMatrix`: If `P.rels` has a `Fintype` instance, we may form
+  the matrix corresponding to `P.differential`. Its determinant is `P.jacobian`.
```

Comment:
What is the content of this field when no `Fintype` instance for `P.rels` is provided?

Reply:
OK, I realize that this is not a field of the structure. Maybe clarify what is part of the structure and what are additional declarations?

Reply:
Is it clearer now?